src/Runner.Listener/Configuration/ConfigurationManager.cs

```diff
@@ -275,6 +275,23 @@ public async Task ConfigureAsync(CommandSettings command)
                 throw new NotSupportedException("Message queue listen OAuth token.");
             }
 
+            // Configure to use fips-compliant encryption schemes for communication if required by server
+            var requireFipsCryptography = agent.Properties.GetValueOrDefault("RequireFipsCryptography") as bool?;
+            runnerSettings.RequireFipsCryptography = requireFipsCryptography ?? false;
```

Comment:
should we store these as part of `credentialData` instead of `runnerSettings`?

Reply:
I assume you don't need most of these change you if store it into `credentialData`?

Reply:
It doesnt seem like there would be much benefit to changing it. We still need to read the setting from the server response (agent properties) since the server will dictate what mode to use

So instead of saving it in the runner settings, it could be saved in the credentialData, but then in the messagelistener to get that data back out, I think we need to get the credentialData out of the settings store anyways

Reply:
I am not sure whether we should change the command line prompt during config, I still prefer to store it into the `credentialData.Data` dictionary in line 265.

you can get the data out in MessageListener by 
```
var configStore = HostContext.GetService<IConfigurationStore>();
var credData = configStore.GetCredentials();
var requireFipsCryptography = StringUtil.ConvertToBoolean(credData.Data.GetValueOrDefault("requireFipsCryptography"), false);                    
```

So we don't need to change the runner setting at all to minimize the overall code changes.

Reply:
👍  I updated it to store that setting in credentialData.